trie/verkle.go

```diff
@@ -217,22 +217,16 @@ func (t *VerkleTrie) Hash() common.Hash {
 }
 
 // Commit writes all nodes to the tree's memory database.
-func (t *VerkleTrie) Commit(_ bool) (common.Hash, *trienode.NodeSet, error) {
-	root, ok := t.root.(*verkle.InternalNode)
-	if !ok {
-		return common.Hash{}, nil, errors.New("unexpected root node type")
-	}
-	nodes, err := root.BatchSerialize()
-	if err != nil {
-		return common.Hash{}, nil, fmt.Errorf("serializing tree nodes: %s", err)
-	}
+func (t *VerkleTrie) Commit(_ bool) (common.Hash, *trienode.NodeSet) {
+	root := t.root.(*verkle.InternalNode)
+	nodes, _ := root.BatchSerialize()
```

Comment:
I think that this doesn't need to return an error, and opened an upstream [PR](https://github.com/ethereum/go-verkle/pull/435).

Reply:
If it doesn't get accepted, we can just ignore the error return here IMO.  afaict, indicates an internal logic error that we should not experience under normal operation and can't reasonably recover from.

Reply:
Mazzika

Reply:
Mazzika

Reply:
Added a panic if an error is returned here.  We assume by this point, that the `Trie` we are committing is well-formed.

TBQH, I don't fully understand the underlying logic of what is happening here, but it seems clear that this can only return an error if the underlying tree is somehow corrupted or the code logic is wrong, both of which should be unrecoverable IMO.  The underlying error originates [here](https://github.com/ethereum/go-verkle/blob/7d920df305f0f1fc135f3ed70443263c5fa62cf9/tree.go#L1753).

CC @gballet if there is something I'm missing here.